### PR TITLE
Clarify merge requirements

### PR DESCRIPTION
**Pull request by mosabua**
Also note that I think we should add more info on what happens afterwards. I can do that in a follow up PR or in this one .. 

Basically 

* Maintainers and dev rel team organize release note entry for PR.
* Maintainers initiate release when possible, Aim is to release once a week. 
* The release gets the changes from the PR to users in the forms of updates binaries and documentation.

Wdyt @martint .. should I address the above points in this PR as well? Or a follow up?

### CODE REVIEW

File: development/process.md
Comment (martint) [inline development/process.md:86]: One maintainer approving is sufficient.

Comment (mosabua) [reply development/process.md:86]: Well.. yes.. but the total is always two .. the maintainer who merges.. and one other... that was my misunderstanding. Not sure how to word this better yet.

Comment (martint) [reply development/process.md:86]: Actually, no. If it’s a contribution from someone who’s not a maintainer it only requires one maintainer to review and merge

Comment (mosabua) [reply development/process.md:86]: oh .. so the exception is the other way around ... I will update tomorrow after I figure out some clear wording for that.

Comment (martint) [reply development/process.md:86]: There’s no exception. Any contribution needs to be reviewed by a maintainer that wasn’t the author of the contribution

Comment (mosabua) [reply development/process.md:86]: Yeah ... "that wasn't the author of the contribution" is the key here.

**Comment by mosabua** [comment]
I added what happens after merge so contributors know what to expect as well.

File: development/process.md
Comment (mosabua) [reply development/process.md]: I think I addressed this in the wording now.

File: development/process.md
Comment (mosabua) [reply development/process.md:105]: Maybe we should drop this part .. essentially I want to say that now it goes up to the top of the document and the same work continues with all sorts of other PRs.. but maybe thats redundant.

File: development/process.md
Comment (colebow) [inline development/process.md:86]: The wording on this is a little awkward, and I'm actually not sure if it's saying you need two maintainers when one is an author, or if you need one maintainer period. If the latter, maybe just "A maintainer can merge a pull request that they authored as long as it is approved by another reviewer."

File: development/process.md
Comment (colebow) [inline development/process.md:88]:
```suggestion
    * Approval of a reviewing maintainer is required before a pull request can be
      merged. If a maintainer authors a pull request, another maintainer must
      approve it.
```

**Comment by mosabua** [comment]
Clever rewording @colebow  .. pulled in now

File: development/process.md
Comment (mosabua) [reply development/process.md]: Dropped .. its cleaner that way.

File: development/process.md
Comment (martint) [inline development/process.md:100]: Missing period at end of sentence

**Review by martint — APPROVED** [review]

File: development/process.md
Comment (mosabua) [reply development/process.md:100]: Fixed

**Pull request merged**
